Command substitution from commit message in project-upload action

## Summary

`project-upload@v1.1.0` interpolates `github.event.head_commit.message` directly into a bash command string when building the `--message` flag of `hs project upload`. As a result, any backtick or `$(...)` inside a commit message is interpreted by bash as command substitution at deploy time, causing the upload step to fail.

## Reproduction

Any PR whose merge commit body contains backtick-delimited substrings causes the deploy to fail.

Minimal example commit message:

```
chore(ci): add a workflow

Wire the repo to `some-shared-workflow.yml@v1` per pattern in `other-repo`.
```

The two backtick-quoted strings are not arguments to `hs project upload` — they end up being executed as shell commands. The deploy log shows:

```
/home/runner/work/_temp/<id>.sh: line 63: some-shared-workflow.yml@v1: command not found
/home/runner/work/_temp/<id>.sh: line 63: other-repo: command not found
##[error]Process completed with exit code 2.
```

A real failure of this kind in our org: <link to the failing run in our private repo would go here>.

## Root cause

[`project-upload/action.yml`](https://github.com/HubSpot/hubspot-project-actions/blob/v1.1.0/project-upload/action.yml), around line 65:

```bash
run_hs_command \
  "hs project upload --force-create --use-env --json --message '${{ github.event.head_commit.message || 'Uploaded via HubSpot GitHub Action' }} (${GITHUB_SHA:0:7})' $PROFILE_ARG" \
  "true"
```

The whole `hs project upload ...` invocation is built as a single bash string with the commit message text spliced in via `${{ github.event.head_commit.message }}`. When that string is later evaluated by bash, any backticks inside the commit-message text are processed as command substitution. Single quotes around the message don't help because `${{ ... }}` substitution happens at workflow-template render time, *before* bash sees the line, so the resulting line is literally:

```bash
hs project upload --force-create --use-env --json --message 'Wire the repo to `some-shared-workflow.yml@v1` per pattern in `other-repo`. (abc1234)' --profile prod
```

The bash quoting in the resulting line is broken: the single quote after `Wire the repo to ` opens, then the backticks open command substitution, the next single quote closes, etc. The exact behaviour varies by message but the end result is reliably "the deploy fails".

## Suggested fix

Pass the commit message through an environment variable instead of interpolating it into the bash string. Env values are read by bash via parameter expansion (`"$VAR"`), which treats backticks and `$()` inside the value as literal characters.

Approximate patch:

```yaml
    - name: Upload source code to HubSpot Project
      shell: bash
      env:
        HUBSPOT_PERSONAL_ACCESS_KEY: ${{ inputs.personal_access_key || env.DEFAULT_PERSONAL_ACCESS_KEY }}
        HUBSPOT_ACCOUNT_ID: ${{ inputs.account_id || env.DEFAULT_ACCOUNT_ID }}
        HUBSPOT_PROFILE: ${{ inputs.profile || env.DEFAULT_PROFILE }}
        DEBUG_MODE: ${{ inputs.debug || env.DEFAULT_DEBUG}}
        HEAD_COMMIT_MESSAGE: ${{ github.event.head_commit.message || 'Uploaded via HubSpot GitHub Action' }}
      run: |
        . "$GITHUB_ACTION_PATH/../scripts/action-utils.sh"
        validate_account_and_personal_access_key
        resolve_project_dir "${{ inputs.project_dir }}"

        PROFILE_ARG=""
        if [ -n "$HUBSPOT_PROFILE" ]; then
          PROFILE_ARG="--profile $HUBSPOT_PROFILE"
        fi

        # Keep the first line only to avoid multi-line messages confusing the CLI prompt.
        SAFE_MSG="$(printf '%s' "$HEAD_COMMIT_MESSAGE" | head -n 1)"

        run_hs_command \
          "hs project upload --force-create --use-env --json --message \"${SAFE_MSG} (${GITHUB_SHA:0:7})\" $PROFILE_ARG" \
          "true"

        set_output_from_json "build_id" ".buildId"
        set_output_from_json "deploy_id" ".deployId" "true"
```

(`run_hs_command` itself appears to `eval` the string, so the same care needs to apply there: use double-quotes around the `--message` value and resolve the env var via parameter expansion.)

## Workaround for users hitting this today

Trigger the deploy via `workflow_dispatch` instead of relying on `push`. The action's fallback message "Uploaded via HubSpot GitHub Action" is used when `github.event.head_commit` is empty, which it is for that event. This avoids the bug at the cost of a manual click for every deploy.

Replacing the action call with `install-hubspot-cli` + a manual `hs project upload --message "${SAFE_MSG}"` (passing the message via env var) is also a viable workaround.

## Environment

- `HubSpot/hubspot-project-actions/project-upload@v1.1.0`
- GitHub-hosted ubuntu-24.04 runners
- HubSpot CLI as installed by the action's pinned default version

--- a/.github/workflows/test-message-escaping.yml
+++ b/.github/workflows/test-message-escaping.yml
@@ -1,0 +1,73 @@
+name: Test Message Escaping
+
+on: [push, pull_request, workflow_dispatch]
+
+jobs:
+  test-message-escaping:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - name: Test that shell metacharacters in commit messages are preserved
+        shell: bash
+        run: |
+          . ./scripts/action-utils.sh
+
+          # Mock hs CLI: captures the --message value to a file and returns valid JSON
+          hs() {
+            while [[ $# -gt 0 ]]; do
+              case "$1" in
+                --message) echo "$2" > /tmp/captured_message; shift 2 ;;
+                *) shift ;;
+              esac
+            done
+            echo '{"buildId":"test-123","deployId":"test-456"}'
+          }
+          export -f hs
+
+          test_message() {
+            local test_name="$1"
+            local input="$2"
+            local expected="$3"
+
+            HEAD_COMMIT_MESSAGE="$input"
+            UPLOAD_MESSAGE="$(printf '%s' "$HEAD_COMMIT_MESSAGE" | head -n 1) (abc1234)"
+
+            run_hs_command \
+              "hs project upload --force-create --use-env --json --message \"\$UPLOAD_MESSAGE\"" \
+              "true"
+
+            local captured
+            captured=$(cat /tmp/captured_message)
+
+            if [[ "$captured" == *"$expected"* ]]; then
+              echo "PASS: $test_name"
+            else
+              echo "FAIL: $test_name"
+              echo "  Expected to contain: $expected"
+              echo "  Got: $captured"
+              exit 1
+            fi
+          }
+
+          test_message \
+            "backticks are preserved" \
+            'Wire repo to `some-workflow.yml@v1` per pattern in `other-repo`' \
+            '`some-workflow.yml@v1`'
+
+          test_message \
+            "dollar-paren command substitution is not executed" \
+            'Test $(echo pwned) in message' \
+            '$(echo pwned)'
+
+          test_message \
+            "double quotes are preserved" \
+            'Test with "double quotes" inside' \
+            '"double quotes"'
+
+          test_message \
+            "multiline message is truncated to first line" \
+            $'First line\nSecond line\nThird line' \
+            'First line'
+
+          echo ""
+          echo "All message escaping tests passed"

--- a/project-upload/action.yml
+++ b/project-upload/action.yml
@@ -62,6 +62,8 @@ runs:
           PROFILE_ARG="--profile $HUBSPOT_PROFILE"
         fi
 
+        # Build the upload message from the env var (not via ${{ }} template interpolation) so that
+        # shell metacharacters like backticks in the commit message are never executed by bash.
         UPLOAD_MESSAGE="$(printf '%s' "$HEAD_COMMIT_MESSAGE" | head -n 1) (${GITHUB_SHA:0:7})"
 
         # Run upload command

--- a/project-upload/action.yml
+++ b/project-upload/action.yml
@@ -45,6 +45,7 @@ runs:
         HUBSPOT_ACCOUNT_ID: ${{ inputs.account_id || env.DEFAULT_ACCOUNT_ID }}
         HUBSPOT_PROFILE: ${{ inputs.profile || env.DEFAULT_PROFILE }}
         DEBUG_MODE: ${{ inputs.debug || env.DEFAULT_DEBUG}}
+        HEAD_COMMIT_MESSAGE: ${{ github.event.head_commit.message || 'Uploaded via HubSpot GitHub Action' }}
       run: |
         # Source shared utilities
         . "$GITHUB_ACTION_PATH/../scripts/action-utils.sh"
@@ -61,9 +62,11 @@ runs:
           PROFILE_ARG="--profile $HUBSPOT_PROFILE"
         fi
 
+        UPLOAD_MESSAGE="$(printf '%s' "$HEAD_COMMIT_MESSAGE" | head -n 1) (${GITHUB_SHA:0:7})"
+
         # Run upload command
         run_hs_command \
-          "hs project upload --force-create --use-env --json --message '${{ github.event.head_commit.message || 'Uploaded via HubSpot GitHub Action' }} (${GITHUB_SHA:0:7})' $PROFILE_ARG" \
+          "hs project upload --force-create --use-env --json --message \"\$UPLOAD_MESSAGE\" $PROFILE_ARG" \
           "true"
 
         # Parse and set outputs

